ddtrace/debugging/_debugger.py

```diff
@@ -426,28 +421,12 @@ def _dd_debugger_hook(self, probe: Probe) -> None:
                 log.error("Unsupported probe type: %r", type(probe))
                 return
 
-            scope = ChainMap(actual_frame.f_locals, actual_frame.f_globals)
+            signal.do_line()
 
-            try:
-                if isinstance(probe, ProbeConditionMixin) and not signal._eval_condition(scope):
-                    return
-
-                if isinstance(probe, RateLimitMixin) and probe.limiter.limit() is RateLimitExceeded:
-                    signal.state = SignalState.SKIP_RATE
-                    return
-
-                try:
-                    signal.line(scope)
-                except Exception:
-                    log.exception("Failed to execute line probe %r", probe)
-                    return
-
-                signal.state = SignalState.DONE
+            self._probe_registry.set_emitting(probe)
```

Comment:
I think we miss a check if signal.state is SignalState.DONE before set emitting.